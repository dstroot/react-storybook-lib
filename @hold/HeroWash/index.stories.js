import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";

import HeroWash from "../HeroWash";
import logo from "../HeroWash/media/PacificLife_BrandIcon_White.png";
import image from "../HeroWash/media/2.jpg";

const stories = storiesOf("HeroWash", module);
stories.addDecorator(withKnobs);

stories
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false
    }
  })
  .add("show background 1", () => (
    <HeroWash
      logo={logo}
      bgimage={image}
      tagline={text("Tagline", "You. Protected.")}
      line1={text("Line 1", "Our mission is protecting your financial future.")}
      line2={text("Line 2", "We’ve fulfilled that promise for 150 years.")}
    />
  ))
  .add("show background 2", () => (
    <HeroWash
      logo={logo}
      bgimage={text(
        "Background Image",
        "https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg"
      )}
      tagline={text("Tagline", "You. Protected.")}
      line1={text("Line 1", "Our mission is protecting your financial future.")}
      line2={text("Line 2", "We’ve fulfilled that promise for 150 years.")}
    />
  ));
