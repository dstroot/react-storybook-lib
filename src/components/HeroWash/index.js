import React from "react";
import PropTypes from "prop-types";

/**
HeroWash is a component that applies a primary color wash over a background image. 
It is fully responsive and even the text uses fluid sizes.
 */
const HeroWash = ({ logo, bgimage, tagline, line1, line2 }) => {
  const divHero = {
    /* Image with a color wash */
    backgroundImage: `linear-gradient(to top, rgba(0, 142, 215, 0.9), rgba(0, 142, 215, 0.4)), url(${bgimage})`,

    /* Set height relative to width */
    height: "40vw",
    minHeight: "325px",

    /* Position and center the image to scale nicely on all screens */
    backgroundPosition: "center",
    backgroundRepeat: "no-repeat",
    backgroundSize: "cover",
    position: "relative"
  };

  // place the text on the hero image - center text, viewport wide, at the bottom
  const heroText = {
    textAlign: "center",
    position: "absolute",
    bottom: "5vw",
    width: "100vw",
    color: "white"
  };

  const textBig = {
    fontSize: `calc(32px + 1.5vw + 0.5vmin)`,
    marginBottom: "0",
    fontWeight: "500"
  };

  const textSmall = {
    marginTop: "-0.5vw",
    marginBottom: "0",
    fontSize: `calc(8px + 1.5vw + 0.5vmin)`,
    fontWeight: "300"
  };

  return (
    <div style={divHero}>
      <div style={heroText}>
        <img className="d-block-inline" src={logo} alt="hero" />
        <p style={textBig}>{tagline}</p>
        <p style={textSmall}>{line1}</p>
        <p style={textSmall}>{line2}</p>
      </div>
    </div>
  );
};

// note the comments are used for storybook.
HeroWash.propTypes = {
  /** string representing a URL for the background image */
  bgimage: PropTypes.string.isRequired,
  /** Additional line 1 */
  line1: PropTypes.string,
  /** Additional line 2 */
  line2: PropTypes.string,
  /** string representing a URL for the logo */
  logo: PropTypes.string.isRequired,
  /** string for the tagline */
  tagline: PropTypes.string.isRequired
};

export default HeroWash;
