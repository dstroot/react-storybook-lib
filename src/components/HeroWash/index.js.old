import React from 'react';
import test from '../../media/PacificLife_BrandIcon_White.svg';

const HeroWash = ({ image, tagline, line1, line2 }) => {
  const divHero = {
    /* Image with a color wash */
    backgroundImage: `linear-gradient(to top, rgba(0, 142, 215, 0.9), rgba(0, 142, 215, 0.4)), url(${image})`,

    /* Set height relative to width */
    height: '40vw',
    minHeight: '225px',

    /* Position and center the image to scale nicely on all screens */
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    position: 'relative',
  };

  // place the text on the hero image - center text, viewport wide, at the bottom
  const heroText = {
    textAlign: 'center',
    position: 'absolute',
    bottom: '5vw',
    width: '100vw',
    color: 'white',
  };

  const textBig = {
    fontSize: `calc(32px + 1.5vw + 0.5vmin)`,
    marginBottom: '0',
    fontWeight: '500',
  };

  const textSmall = {
    marginTop: '-0.5vw',
    marginBottom: '0',
    fontSize: `calc(8px + 1.5vw + 0.5vmin)`,
    fontWeight: '300',
  };

  return (
    <div style={divHero}>
      <div style={heroText}>
        <img className="d-block-inline" src={test} alt="" />
        <p style={textBig}>{tagline}</p>
        <p style={textSmall}>{line1}</p>
        <p style={textSmall}>{line2}</p>
      </div>
    </div>
  );
};

export default HeroWash;
