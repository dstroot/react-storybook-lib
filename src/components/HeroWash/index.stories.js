import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import HeroWash from '../HeroWash';
import logo from '../HeroWash/media/PacificLife_BrandIcon_White.png';
import image from '../HeroWash/media/2.jpg';

const stories = storiesOf('Pac Life|HeroWash', module);
stories.addDecorator(withKnobs);

stories
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `
      You. Protected.
      Our mission is protecting your financial future.
      Weâ€™ve fulfilled that promise for 150 years.
      `,
    },
  })
  .add('show background 1', () => (
    <HeroWash
      logo={logo}
      bgimage={image}
      tagline={text('Tagline', 'Tagline')}
      line1={text('Line 1', 'Some additional text')}
      line2={text('Line 2', 'More additional text')}
    />
  ))
  .add('show background 2', () => (
    <HeroWash
      logo={logo}
      bgimage={text(
        'Background Image',
        'https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg'
      )}
      tagline={text('Tagline', 'Tagline')}
      line1={text('Line 1', 'Some additional text')}
      line2={text('Line 2', 'More additional text')}
    />
  ));
