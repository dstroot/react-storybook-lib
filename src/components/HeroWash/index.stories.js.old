import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import HeroWash from '../HeroWash';

// images
import logo from '../HeroWash/media/PacificLife_BrandIcon_White.png';
import image from '../HeroWash/media/2.jpg';

storiesOf('HeroWash', module)
  .add('show HeroWash image 1', () => (
    <HeroWash
      logo={logo}
      bgimage="https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg"
      tagline="You. Protected."
      line1="Our mission is protecting your financial future."
      line2="We’ve fulfilled that promise for 150 years."
    />
  ))
  .add('show HeroWash image 2', () => (
    <HeroWash
      logo={logo}
      bgimage={image}
      tagline="You. Protected."
      line1="Our mission is protecting your financial future."
      line2="We’ve fulfilled our promise for 150 years."
    />
  ));
