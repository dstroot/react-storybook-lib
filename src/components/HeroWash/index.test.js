import React from 'react';
import renderer from 'react-test-renderer';

import HeroWash from '../HeroWash';

test('HeroWash renders some content', () => {
  const component = renderer.create(
    <HeroWash
      logo=""
      bgimage="https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg"
      tagline="You. Protected."
      line1="Our mission is protecting your financial future."
      line2="Weâ€™ve fulfilled our promise for 150 years."
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
