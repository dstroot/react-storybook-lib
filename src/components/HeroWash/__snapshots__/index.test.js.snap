// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroWash renders some content 1`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to top, rgba(0, 142, 215, 0.9), rgba(0, 142, 215, 0.4)), url(https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "height": "40vw",
      "minHeight": "325px",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "bottom": "5vw",
        "color": "white",
        "position": "absolute",
        "textAlign": "center",
        "width": "100vw",
      }
    }
  >
    <img
      alt="hero"
      className="d-block-inline"
      src=""
    />
    <p
      style={
        Object {
          "fontSize": "calc(32px + 1.5vw + 0.5vmin)",
          "fontWeight": "500",
          "marginBottom": "0",
        }
      }
    >
      You. Protected.
    </p>
    <p
      style={
        Object {
          "fontSize": "calc(8px + 1.5vw + 0.5vmin)",
          "fontWeight": "300",
          "marginBottom": "0",
          "marginTop": "-0.5vw",
        }
      }
    >
      Our mission is protecting your financial future.
    </p>
    <p
      style={
        Object {
          "fontSize": "calc(8px + 1.5vw + 0.5vmin)",
          "fontWeight": "300",
          "marginBottom": "0",
          "marginTop": "-0.5vw",
        }
      }
    >
      Weâ€™ve fulfilled our promise for 150 years.
    </p>
  </div>
</div>
`;
